c4200m_get_target_sentences.py

```diff
@@ -5,44 +5,75 @@
 
 from absl import app
 import tensorflow_datasets as tfds
+import multiprocessing as mp
+import re
 
 LOGGING_STEPS = 100000
+BATCH_SIZE = 64
 
+def rreplace(string, old, new):
+  return new.join(string.rsplit(old, 1))
+
+def get_file_paths(edits_tsv_path, concurrent_runs):
+  start_index_string = re.search('(\d+)', edits_tsv_path).group(0)
+  start_index = int(start_index_string)
+  end_index = min(start_index + concurrent_runs, 10)
+  return [rreplace(edits_tsv_path, start_index_string, start_index_string[:-1] + str(i)) 
+            for i in range(start_index, end_index)]
+
+def load_file(tup):
+  index, edits_tsv_path = tup
+  print("Loading C4_200M target sentence hashes from %r..." % edits_tsv_path)
+  with open(edits_tsv_path, "r", encoding="utf-8") as edits_tsv_reader:
+    return { tsv_line.split("\t", 1)[0]: index for tsv_line in edits_tsv_reader}
+        
+def save_file(tup):
+  target_sentences, output_tsv_path = tup
+  print("Writing C4_200M sentence pairs to %r..." % output_tsv_path)
+  with open(output_tsv_path, "w", encoding="utf-8") as output_tsv_writer:
+    while target_sentences:
+      output_tsv_writer.write("%s\t%s\n" % heapq.heappop(target_sentences))
 
 def main(argv):
-  if len(argv) != 3:
+  if len(argv) < 3:
     raise app.UsageError(
-        "python3 c4200m_get_target_sentences.py <edits-tsv> <output-tsv>")
+        "python3 c4200m_get_target_sentences.py <edits-tsv> <output-tsv> [<concurrent-runs>]")
   edits_tsv_path = argv[1]
   output_tsv_path = argv[2]
+  concurrent_runs = min(int(argv[3]), 10) if len(argv) > 3 else 1
+  
+  edits_tsv_paths = get_file_paths(edits_tsv_path, concurrent_runs)
+  output_tsv_paths = get_file_paths(output_tsv_path, concurrent_runs)
 
-  print("Loading C4_200M target sentence hashes from %r..." % edits_tsv_path)
-  remaining_hashes = set()
-  with open(edits_tsv_path) as edits_tsv_reader:
-    for tsv_line in edits_tsv_reader:
-      remaining_hashes.add(tsv_line.split("\t", 1)[0])
+  remaining_hashes = dict()
```

Comment:
why changing it from a set to a dict?

Reply:
since we are going through multiple files, we need to store the index of the original file in the list. i.e. `"hash": 0`